app/controllers/application_controller.rb

```diff
@@ -6,4 +6,11 @@ class ApplicationController < ActionController::Base
     def current_user
         @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
     end
+
+    def authenticate_user
+        if current_user == nil
+          flash[:notice] = "ログインが必要です"
+          redirect_to("/login")
+        end
```

Comment:
```suggestion
redirect "/login", notice: "ログインが必要です" if current_user.blank?
```

このぐらいならワンライナーで書いた方がすっきり見えそうです！

あと変数やメソッドの返り値が空かどうかをチェックするのはblank?という便利メソッドがあるので使えるようになルと良さそうです！
https://pikawaka.com/rails/blank